include/deal.II/non_matching/mapping_info.h

```diff
@@ -1447,7 +1611,7 @@ namespace NonMatching
   MappingInfo<dim, spacedim, Number>::get_unit_point(
     const unsigned int offset) const
   {
-    return &unit_points[offset];
+    return unit_points.data() + offset;
```

Comment:
I like this change. But I hope the compiler creates the same code for both cases!?

Reply:
This change was motivated by the code in `do_reinit()` that sets the pointers unconditionally in release mode. Otherwise you get segfaults if entry 0 does not exist.